src/plugins/gds_mt/gds_mt_backend.cpp

```diff
@@ -0,0 +1,296 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 Serapheim Dimitropoulos, WekaIO Ltd.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cassert>
+#include <iostream>
+#include <cufile.h>
+#include <thread>
+#include <memory>
+#include "common/nixl_log.h"
+#include "gds_mt_backend.h"
+#include "common/str_tools.h"
+
+namespace {
+    const size_t DEFAULT_THREAD_COUNT = std::thread::hardware_concurrency();
```

Comment:
Seems using all hardware threads is not an optimal performance option, may be a fraction might be better?

Reply:
I'm halving the threads for now (see updated code) - let me know if you'd prefer something different.